src/main/webapp/gast/suche/freie_suche.jsp

```diff
@@ -284,13 +292,18 @@
     }
     conditions.add("quelle.zuVeroeffentlichen=1");
     einzelbeleg = true;
-    if (!request.getParameter("Quelle").trim().equals("") && request.getParameter("Quellenliste").equals("-1")) {
+    if (!request.getParameter("Quelle").trim().equals("") && (request.getParameterValues("Quellenliste[]") == null)) {
         conditions.add("quelle.Bezeichnung LIKE '" + request.getParameter("Quelle").trim() + "'");
         namenkommentar = true;
     }
-    if (!request.getParameter("Quellenliste").equals("-1")) {
-        conditions.add("quelle.ID=" + request.getParameter("Quellenliste"));
-        einzelbeleg = true;
+    String[] quellenliste;
+    if ((quellenliste = request.getParameterValues("Quellenliste[]")) != null) {
+        for (String qid : quellenliste) {
+            if (!qid.equals("-1")) {
+                orConditions.add("quelle.ID=" + qid);
```

Comment:
Don't forget to escape all parameters that are taken from the request to build MySQL statements, else we are vulnerable to SQL Injection